src/content/docs/recipes/handling-dark-mode.mdx

```diff
@@ -0,0 +1,162 @@
+---
+title: Handling dark mode
+---
+
+import { Tabs, TabItem, Steps } from "@astrojs/starlight/components";
+
+Adding dark mode support to improve the user experience is now a common need on the web. And surprisingly, it can proved difficult to get it right using Astro. Let's see how to do it the right way!
+
+## Solution
+
+<Steps>
+
+1. Getting the current theme from local storage or system preference
+
+    ```ts title="src/components/theme-toggle/script.js"
+    /** @type {'dark'|'light'|'auto'} */
+    let theme = ('theme' in localstorage) 
+        ? localstorage.theme 
+        : window.matchmedia('(prefers-color-scheme: dark)').matches 
+            ? 'dark' 
+            : 'light'
+    ```
+
+2. Define a Setter function to store the theme value in local storage
+
+    ```ts title="src/components/theme-toggle/script.js"
+    /**
+    * @param {'dark'|'light'|'auto'} theme
+    */
+    const setTheme = (theme = 'auto') => {
+        localStorage.setItem('theme', theme);
+        theme = theme;
+    }
+    ```
+
+3. if the theme is "dark", add the "dark" class to the `html` element, else remove the "dark" class from `html` element
+
+    ```ts title="src/components/theme-toggle/script.js"
+    /**
+    * @param {'dark'|'light'|'auto'} theme
+    */
+    const setTheme = (theme = 'auto') => {
+        localStorage.setItem('theme', theme);
+        theme = theme;
+        if (theme === 'dark') 
```

Comment:
I think it would be worth adopting the tailwind script strategy I showed earlier. Set the data attribute and later on a mutation observer toggles the `dark` class based on the data attribute. This way, people relying only on the data attribute can get rid of the mutation observer

Reply:
@OliverSpeir what do you think?